### PR TITLE
ceph: 14.2.4 -> 14.2.5

### PR DESCRIPTION
**Pull request by srhb**
###### Motivation for this change

Fixes a critical integrity bug n 14.2.3 and 14.2.4. Adds newly required deps on libnl and libcap_ng.

This needs backporting to 19.09, but I think we should also backport the various fixes to rocksdb etc. before doing so. I can author that.

WIP while my poor laptop chugs away at building it...

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by srhb** [comment]
Ready for review. I'll hopefully be able to deploy this to a test cluster soonish, but I think it's good from testing in VMs.

There's a critical bug in the current 14.2.4 version which may cause OSD corruption when bluestore WALs are on separate devices.
There are other, smaller changes to deps and behaviour (specifically more health checks) that required tests to be tuned slightly.

Note that the multi node test currently fails because of a regression in the python test suite which causes files to no longer be shared among VMs in `/tmp/shared`. Reverting ef5bc381f969205493897b121ce36ecd7bdc1974 causes the ceph test to succeed, so I think we're good from this side.

**Comment by srhb** [comment]
Oh, and I'll probably set up a full backport of floklis fixes to ceph/rocksdb and this, if this passes review, since we really need to get this version in stable as well, and the changes should be safe. :)

File: nixos/tests/ceph-multi-node.nix [hunk 1]
```diff
@@ -22,6 +22,8 @@ let
   };
   generateCephConfig = { daemonConfig }: {
     enable = true;
+    # 3 is the default, and we have only 2, so we get HEALTH_WARN
+    extraConfig."osd pool default size" = "2";
```
Comment (flokli) [inline nixos/tests/ceph-multi-node.nix]: We might just want to increase the number of OSDs, by providing another VM for it.

**Comment by flokli** [comment]
cc @7c6f434c w.r.t `/tmp/shared`.

File: nixos/tests/ceph-multi-node.nix (hunk 1 above)
Comment (srhb) [reply nixos/tests/ceph-multi-node.nix]: I considered it. It seems a bit wasteful, but on the other hand, no catastrophe. I can do that :)

**Comment by 7c6f434c** [comment]
Hm, maybe we should split the shared reference into internal_shared and external_shared then?

File: nixos/tests/ceph-multi-node.nix (hunk 1 above)
Comment (srhb) [reply nixos/tests/ceph-multi-node.nix]: Fixed.

**Review by lejonet — APPROVED** [review]
I can't really see anything wrong with the changes. Thumbs up from me!

**Pull request merged**